examples/medplum-provider/src/components/ResourceFormWithRequiredProfile.tsx

```diff
@@ -0,0 +1,78 @@
+import { Alert } from '@mantine/core';
+import { InternalTypeSchema, normalizeErrorString, tryGetProfile } from '@medplum/core';
+import { Loading, ResourceForm, ResourceFormProps, useMedplum } from '@medplum/react';
+import { IconAlertCircle } from '@tabler/icons-react';
+import { ReactNode, useCallback, useEffect, useState } from 'react';
+import { addProfileToResource } from '../utils';
+import { Resource } from '@medplum/fhirtypes';
+
+interface ResourceFormWithRequiredProfileProps extends ResourceFormProps {
+  /** (optional) If specified, an error is shown in place of `ResourceForm` if the profile cannot be loaded.  */
+  profileUrl?: string; // Also part of ResourceFormProps, but list here incase its type changes in the future
```

Comment:
Sonar would tell you to make this `readonly`

Currently all `examples` are excluded from Sonar.  Not sure if you want to enable that?